### PR TITLE
Auth service for client created

### PR DESCRIPTION
**Pull request by YuraRiabov**
Sign up, sign in with email/password, sign in with google, sign out, checking if user is logged in and password reset implemented using firebase

### CODE REVIEW

File: frontend/src/app/core/services/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public signOut() {
+        return this.afAuth.signOut().then(() => localStorage.removeItem('user'));
+    }
+
+    public isLoggedIn() {
+        const currentUser = localStorage.getItem('user');
+
+        if (currentUser === null) {
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts]:
```suggestion
        if (!currentUser) {
```

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:1]: Could you please add error handling for to each method (that needs it)?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/.editorconfig [hunk 3]
```diff
@@ -4,7 +4,7 @@ root = true
 [*]
 charset = utf-8
 indent_style = space
-indent_size = 2
+indent_size = 4
```
Comment (apofis96) [inline frontend/.editorconfig:7]: Leave it 2. It's frontend standard (de facto)

File: frontend/src/environments/environment.prod.ts [hunk 4]
```diff
@@ -2,4 +2,13 @@ export const environment = {
     production: true,
     coreUrl: '/api',
     notifierUrl: '/notifier',
+    firebase: {
```
Comment (apofis96) [inline frontend/src/environments/environment.prod.ts:5]: It's a bad idea to commit security data to repo. Leave it with empty values.

File: frontend/src/environments/environment.ts [hunk 5]
```diff
@@ -6,6 +6,15 @@ export const environment = {
     production: false,
     coreUrl: 'http://localhost:5050',
     notifierUrl: 'http://localhost:5070',
+    firebase: {
```
Comment (apofis96) [inline frontend/src/environments/environment.ts:9]: Same

**Review by tatianahutii — APPROVED** [review]

**Comment by tatianahutii** [comment]
LGTM

**Pull request merged**